sklearn/decomposition/_dict_learning.py

```diff
@@ -1324,16 +1334,14 @@ def __init__(
         )
         self.dictionary = dictionary
 
+    @_fit_context(prefer_skip_nested_validation=True)
     def fit(self, X, y=None):
-        """Do nothing and return the estimator unchanged.
-
-        This method is just there to implement the usual API and hence
-        work in pipelines.
+        """Validate the input data.
```

Comment:
The docstring of fit of other stateless estimators looks like the following. Let's be consistent
```suggestion
        """Only validate the parameters of the estimator.

        This method allows to: (i) validate the parameters of the estimator and
        (ii) be consistent with the scikit-learn transformer API.
```